### DISCUSSION TITLE
How to switch Stripe accounts after user connected/merchant accounts have already been created

### DISCUSSION THREAD

**Discussion by Youdaman** [pay-rails/pay#1020] [Q&A]
Not sure if this is something that can be solved on the Pay gem side of things or requires Stripe API kung fu.

When I updated my stripe private/public keys in the credentials file (switching from my AU Stripe account to my US Stripe account) I got the following error when trying to generate a login link for users with `current_user.merchant_processor.login_link`:
```
The provided key 'sk_test_*********************************************************************************************yvvs1C' does not have access to account 'acct_1PUI8wPp104M9HJT' (or that account does not exist). Application access may have been revoked.
```

I was hoping that pay_merchant/user accounts would be able to just switch to using the new platform, but it seems they're associated/accessible only with the platform Stripe account that created them (which makes sense I guess).

Not sure what my best way forward here is other than to somehow wipe/reset the pay_merchant data for each user and have them sign up fresh with the new platform, essentially creating new connected accounts and abandoning the old ones, or whether there's a way to let the existing connected accounts operate under the new platform.

According to this Stripe support page, Connected accounts cannot be copied over, only customers: https://support.stripe.com/questions/data-that-can-be-copied-between-stripe-accounts

> Individual charges, invoices, plans and subscriptions, coupons, events, logs, SEPA sources, Bacs PaymentMethods, Connected accounts, and Guest customers do not copy over.

Wondering whether there's a way where Pay detects it's hooked up to a different account and proceed accordingly by making users "re-connect"/onboard, but then I guess that opens a whole other can of worms with allowing multiple pay_merchant account details per user which is likely beyond scope.

Any advice would be appreciated!

**Comment by excid3** [comment]
You can clear out the Pay::Merchant records in the database to reset them so they'd need to reconnect to the new Stripe account.

**Comment by Youdaman** [comment]
I opened the Rails console and did a `Pay::Merchant.destroy_all` and I'm now getting an error:

"Missing required param: account." when trying to call `account_link` on the current usrer's `merchant_processor` -- this is how I previously checked if a user had not onboarded yet and it would show a message with a link to the onboarding process.

```ruby
    unless current_user.merchant_processor.onboarding_complete?
      account_link = current_user.merchant_processor.account_link refresh_url: games_url, return_url: games_url
      flash.now[:alert] = "To enable sharing of games and receiving payments from players, please <a href='#{account_link.url}'>onboard with Stripe</a>.".html_safe
    end
```

**Comment by Youdaman** [comment]
Deleted my cookies and did another wipe and things seem to work now.

I think the issue might have been my currently logged in user didn't have a date of birth (as I added this field later in development) and so an error elsewhere expecting this to be set caused the account creation not to work properly (I saw the record in Pay::Merchant was missing its processor_id).

Thanks for the tip on how to reset things!
